src/main/java/ca/uhn/fhir/jpa/starter/CollectDataResourceProvider.java

```diff
@@ -20,110 +29,166 @@
+			throw new InternalErrorException("Could not collect data for measure " + theId, e);
+		}
+		return parameters;
+	}
+
+	public FhirContext getContext() {
+		return context;
+	}
+
+	@Autowired
+	public void setContext(FhirContext context) {
+		this.context = context;
+	}
+
+	public IFhirResourceDao<Measure> getDao() {
+		return dao;
+	}
+
+	@Autowired
+	public void setDao(IFhirResourceDao<Measure> dao) {
+		this.dao = dao;
+	}
+
+	public int getPort() {
+		return port;
+	}
+
+	public void setPort(int port) {
+		this.port = port;
+	}
+
+	@EventListener
+	public void onApplicationEvent(ServletWebServerInitializedEvent event) {
+		setPort(event.getWebServer().getPort());
+	}
+
+	private MeasureReport generateMeasureReport(HttpServletRequest req, Bundle bundle, Measure measure) throws IOException {
+		MeasureReport report = new MeasureReport();
+		report.setStatus(MeasureReport.MeasureReportStatus.COMPLETE);
+		report.setType(MeasureReport.MeasureReportType.DATACOLLECTION);
+
+		report.setMeasure(measure.getResourceType().name() + "/"
+				+ measure.getIdentifierFirstRep().getValue());
+
+		report.setEvaluatedResource(bundle.getEntry().stream().filter(Bundle.BundleEntryComponent::hasResource)
+				.map(Bundle.BundleEntryComponent::getResource).peek(r -> {
+					IdType idElement = r.getIdElement();
+					String scheme = req.getScheme();
+					StringBuilder sb = new StringBuilder(scheme).append("://").append(req.getServerName());
+					if (("https".equalsIgnoreCase(scheme) && req.getServerPort() != 443) ||
+							("http".equalsIgnoreCase(scheme) && req.getServerPort() != 80)) {
+						sb.append(":").append(req.getServerPort());
+					}
+
+					String baseUrl = sb.append(req.getServletPath()).toString();
+					idElement.setParts(baseUrl, idElement.getResourceType(), idElement.getIdPart(),
+							idElement.getVersionIdPart());
+
+					r.setIdElement(idElement);
+				}).map(Reference::new).collect(Collectors.toList()));
+		return report;
+	}
+
+	private void generateObservationData(Parameters parameters, Bundle bundle) throws IOException {
+		bundle.getEntry().stream().filter(Bundle.BundleEntryComponent::hasResource)
+				.map(Bundle.BundleEntryComponent::getResource)
+				.forEach(r -> parameters.addParameter().setName(PlirConstants.PARAMETER_NAME_RESOURCE).setResource(r));
+	}
+
+	private String constructUrl(Measure measure, DateParam periodStart,
+			DateParam periodEnd) {
+		ParamPrefixEnum startPrefix = ParamPrefixEnum.GREATERTHAN_OR_EQUALS;
+		if (periodStart.getPrefix() != null) {
+			startPrefix = periodStart.getPrefix();
+		}
+
+		ParamPrefixEnum endPrefix = ParamPrefixEnum.LESSTHAN_OR_EQUALS;
+		if (periodEnd.getPrefix() != null) {
+			endPrefix = periodEnd.getPrefix();
+		}
+
+		if (periodEnd.getPrefix() == null) {
+			periodEnd.setPrefix(ParamPrefixEnum.LESSTHAN_OR_EQUALS);
+		}
+
+		return "http://127.0.0.1:" + port + "/fhir/" + measure.getGroupFirstRep().getPopulationFirstRep().getCriteria()
```

Comment:
This looks like fine , the client accesses the server form with in inside the container